packages/node-runtime-worker-thread/webpack.config.js

```diff
@@ -28,6 +28,8 @@ const config = {
     interruptor: 'commonjs2 interruptor',
     'os-dns-native': 'commonjs2 os-dns-native',
     'system-ca': 'commonjs2 system-ca',
+    // @aws-sdk/client-sts is an optional peer dependency of @aws-sdk/credential-providers
+    '@aws-sdk/client-sts': 'commonjs2 @aws-sdk/client-sts',
```

Comment:
Hmmm ... maybe a TODO ticket to clean this up?

But I don't think we should just make this a webpack external because it doesn't happen to get installed in CI. For Evergreen, we have a script (`scripts/mark-ci-required-optional-dependencies.ts`) to actually mark the "required optional" dependencies as required because we know we _want_ them in CI (just not necessarily on user machines when they run e.g. `npx mongosh`). Maybe a similar approach makes sense here?

As part of the Node.js 24 upgrade project, we'll need to take care of making the `@aws-sdk/credential-providers` package a hard dependency of `service-provider-node-driver`, so this may change as part of that (I don't quite know how), hence the suggestion of a TODO ticket rather than doing something right now